server/src/middleware/editFavourites.middleware.ts

```diff
@@ -0,0 +1,20 @@
+import RequestWithUser from '../interfaces/requestWithUser.interface';
+import { NextFunction, Response } from 'express';
+import UserDoesNotHavePermissionToEditThisFavourtie from '../exceptions/UserDoecNotPerrmissionToEditFavourites';
+import NotAuthorizedException from '../exceptions/NotAuthorizedException';
+
+async function favourtieEditMiddleware(request: RequestWithUser, response: Response, next: NextFunction) {
+  try {
+    const favourieListtId = request.params.userId;
+    const userId = request.user._id;
+        if ( userId.toString() !== favourieListtId.toString() ) {
```

Comment:
favourieListtId -misspeled

userId.toString() !== favourieListtId.toString() - this can be extracted to const